src/binary-reader-interpreter.cc

```diff
@@ -890,8 +868,6 @@ static Result on_elem_segment_function_index_check(uint32_t index,
     return Result::Error;
   }
 
-  table->func_indexes[ctx->table_offset++] =
```

Comment:
This was actually a bug that was caught by the transition -- it shouldn't be updating the table in the "check" callback.